web/src/narrow.js

```diff
@@ -74,20 +74,26 @@ export function reset_ui_state() {
     compose_banner.clear_message_sent_banners();
 }
 
-export function changehash(newhash) {
-    if (browser_history.state.changing_hash) {
+export function changehash(newhash, trigger) {
+    if (browser_history.state.changing_hash && trigger !== "retarget message location") {
         return;
     }
     message_viewport.stop_auto_scrolling();
```

Comment:
Do we still want to do this, or should it be in the `browser_history.set_hash` code path?

Reply:
Yeah, if `browser_history.state.changing_hash` is true, then we've already stopped auto-scrolling.

Updated this to be:
```js
export function changehash(newhash, trigger) {
    if (browser_history.state.changing_hash) {
        // If we retargeted the narrow operation because a message was moved,
        // we want to have the current narrow hash in the browser history.
        if (trigger === "retarget message location") {
            window.location.replace(newhash);
        }
        return;
    }
    message_viewport.stop_auto_scrolling();
    browser_history.set_hash(newhash);
}
```